### ISSUE TITLE
Support plotting 1D optimization landscapes

### ISSUE THREAD

**Issue by kiudee** [kiudee/chess-tuning-tools#34]
Todo
-----

- [ ] Support for log scales
- [ ] How to plot data points?
  - Scatter + "pull lines": Easy to do, could be on a vastly different scales due to noise
- [ ] Show uncertainty regions
- [ ] Strip plot for categoricals

**Comment by AlexisOlson** [comment]
## Various 1D plots for reference
---

![image](https://user-images.githubusercontent.com/10537957/94351373-980d6b80-001d-11eb-942d-cd15ece9dc9f.png)
[Source](https://topotoolbox.wordpress.com/2017/10/24/bayesian-optimization-of-the-mn-ratio/)

![image](https://user-images.githubusercontent.com/10537957/94351389-b07d8600-001d-11eb-95be-3711a71f9cb9.png)
[Source](https://scikit-optimize.github.io/stable/auto_examples/ask-and-tell.html)

![image](https://user-images.githubusercontent.com/10537957/94351421-efabd700-001d-11eb-9191-e63b9da26e30.png)
[Source](https://blog.datadive.net/prediction-intervals-for-random-forests/)

![image](https://user-images.githubusercontent.com/10537957/94351437-09e5b500-001e-11eb-88e2-1dbbfa9c3dca.png)
[Source](https://davidstutz.de/a-short-introduction-to-bayesian-neural-networks/)

![image](https://user-images.githubusercontent.com/10537957/94351444-1bc75800-001e-11eb-85fe-c9442b638f32.png)
[Source](https://www.researchgate.net/figure/Interpolation-of-a-noisy-dataset-by-Gaussian-process-regression_fig16_331396400)

![image](https://user-images.githubusercontent.com/10537957/94351592-22ef6580-0020-11eb-8d9a-10d6fc25021b.png)
[Source](https://www.researchgate.net/figure/Illustration-of-Gaussian-process-regression-in-one-dimension-for-the-target-test_fig1_327613136)

![image](https://user-images.githubusercontent.com/10537957/94351447-2c77ce00-001e-11eb-921f-8828ad42b0d0.png)
[Source](http://dfm.io/george/current/user/model/)

![image](https://user-images.githubusercontent.com/10537957/94351453-3bf71700-001e-11eb-88b6-bd59050a9ce2.png)
[Source](https://www.eracons.com/machine_learning)

![image](https://user-images.githubusercontent.com/10537957/94351493-a14b0800-001e-11eb-94c1-ef57985f5db7.png)
[Source](https://micropore.wordpress.com/2017/02/04/python-fit-with-error-on-y-axis/)

![image](https://user-images.githubusercontent.com/18716778/44839243-da162880-ac70-11e8-8970-c72664637ed0.png)
[Source](https://github.com/SheffieldML/GPy/issues/669)

**Comment by kiudee** [comment]
As a WIP, since I started working on the 1D plotting now (after the big refactor), here is an example plot of what I have in mind:
![1d_plot_draft](https://user-images.githubusercontent.com/564473/132999738-91522aed-749e-4c60-bc53-8b0c7fc89109.png)

Labelling, etc. will still change. Error bars on the matches look nice, but were too busy, and it makes it hard to see the mean process of the model. I went with a micro-sized scatter plot to show the data points.
